### PR TITLE
cli: List supported search criteria for jobs/overview route

### PR DESCRIPTION
**Pull request by okurz**
Output in the openqa-cli help:

```
  # List the latest jobs matching the search criteria
  # supported search criteria: distri, version, flavor, build, test, modules, modules_result, module_re, group, groupid, id
  openqa-cli api --osd jobs/overview groupid=135 distri=caasp version=3.0
```

Related progress issue: https://progress.opensuse.org/issues/103864

### CODE REVIEW

**Comment by coolgw** [comment]
Checked, result is OK.
susetest:/home/openqa/openQA/script # ./openqa-cli api --help
Usage: openqa-cli api [OPTIONS] PATH [PARAMS]

  # Show details for job from localhost
  openqa-cli api jobs/4160811

....

  # List all jobs matching the search criteria
  openqa-cli api --osd jobs groupid=135 distri=caasp version=3.0 latest=1

  # List the latest jobs matching the search criteria
  # supported search criteria: distri, version, flavor, build, test, modules, modules_result, module_re, group, groupid, id
  openqa-cli api --osd jobs/overview groupid=135 distri=caasp version=3.0

**Review by coolgw — APPROVED** [review]

File: lib/OpenQA/Constants.pm [hunk 1]
```diff
@@ -98,6 +98,11 @@ use constant VIDEO_FILE_NAME_REGEX => qr/^.*\/video\.[^\/]*$/;
 
 use constant FRAGMENT_REGEX => qr'(#([-?/:@.~!$&\'()*+,;=\w]|%[0-9a-fA-F]{2})*)*';
 
+use constant JOBS_OVERVIEW_SEARCH_CRITERIA => (
+    'distri', 'version', 'flavor', 'build', 'test', 'modules',
+    'modules_result', 'module_re', 'group', 'groupid', 'id'
+);
```
Comment (Martchus) [inline lib/OpenQA/Constants.pm:104]: One could use `qw(…)` here.

**Review by Martchus — APPROVED** [review]

File: lib/OpenQA/CLI/api.pm [hunk 2]
```diff
@@ -83,6 +87,7 @@ sub command {
     openqa-cli api --osd jobs groupid=135 distri=caasp version=3.0 latest=1
 
     # List the latest jobs matching the search criteria
+    # supported search criteria: $search_criteria
```
Comment (kalikiana) [inline lib/OpenQA/CLI/api.pm:90]: This is a comment... with a perl variable?

Comment (kalikiana) [reply lib/OpenQA/CLI/api.pm:90]: Is this expanded via asciidoc or some such?

Comment (okurz) [reply lib/OpenQA/CLI/api.pm:90]: It's not really a comment, it's a POD section where someone used the custom formatting with just a "#" in front of the command examples. The variable is replaced in https://github.com/os-autoinst/openQA/pull/4402/files#diff-4852ca9e62b5150bacb99e7492d2146fdd3af1cbcd4498087dcc9410260e2f8cR15

File: lib/OpenQA/Constants.pm (hunk 1 above)
Comment (okurz) [reply lib/OpenQA/Constants.pm]: done

**Review by kalikiana — APPROVED** [review]

**Pull request merged**